public/js/actions/ArticleActions.js

```diff
@@ -0,0 +1,44 @@
+import dispatcher from '../dispatcher.js'
+
+export default class ArticleActions {
+    // TODO: Have different fetch calls for top/hot articles
+    static fetchArticles() {
+        return new Promise((resolve, reject) => {
+            fetch(`articles`)
+                .then(res => res.json())
+                .then((articles) => {
+                    articles = articles.sort((a, b) => b.votes - a.votes)
+                    dispatcher.dispatch({
+                        articles,
+                        type: 'UPDATE_ARTICLES'
```

Comment:
Changed or unchanged, there is a new article set sent from the server, which will update the articles in the store. And by having a dispatcher it will notify all the stores about the action